src/sage_interface.h

```diff
@@ -0,0 +1,167 @@
+/* LibGAP - a shared library version of the GAP kernel
+ * Copyright (C) 2013 Volker Braun <vbraun.name@gmail.com>
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA. 
+ */
+
+#ifndef LIBGAP__H
+#define LIBGAP__H
+
+
+/*************************************************************************/
+/*** Initialize / Finalize ***********************************************/
+/*************************************************************************/
+
+/* To setup libGAP, you must call libgap_initialize() to
+ * set argv and env
+ *
+ * Two mandatory component of argv is "-K", "memory_pool_size" to set the
+ * libgap memory pool size. For example,
+ *
+ * argc = 5
+ * argv = "libgap", "-l", "/path/to/gap-library", "-K", "64M", NULL
+ * env = NULL
+ */
+void libgap_initialize(int argc, char** argv);
+
+
+/* Currently, this does nothing. One day, it should correctly
+ * finalize libgap so that it can be reinitialized with a different
+ * memory pool...
+ */
+void libgap_finalize(void);
+
+
+/*************************************************************************/
+/*** Local (per-function) Initialization  ********************************/
+/*************************************************************************/
+
+/* You must call libgap_mark_stack_bottom() in every function that
+ * calls into the libGAP C functions. The reason is that the GAP
+ * memory manager will automatically keep objects alive that are
+ * referenced in local (stack-allocated) variables. While convenient,
+ * this requires to look through the stack to find anything that looks
+ * like an address to a memory bag.
+ *
+ * I suggest you use libgap_enter / libgap_exit (see below) instead of
+ * calling libgap_mark_stack_bottom manually.
+ */
+
+extern void* StackBottomBags;
```

Comment:
Perhaps reformat this file (and other changes) using `clang-format` resp. `git clang-format` ?